File: src/main/java/com/ryderbelserion/discord/api/options/types/TokenOption.java
package com.ryderbelserion.discord.api.options.types;

import com.ryderbelserion.discord.api.options.interfaces.IOption;
import joptsimple.OptionParser;
import joptsimple.OptionSet;
import org.jetbrains.annotations.NotNull;
import java.util.Optional;

public class TokenOption extends IOption<String> {

    public TokenOption(@NotNull final OptionParser parser) {
        super(parser);
    }

    @Override
    public @NotNull final Optional<String> getValue(@NotNull final OptionSet option) {
        return Optional.ofNullable(option.valueOf(getName()).toString());
    }

    @Override
    public @NotNull final String getDescription() {
        return "Provides a discord bot token to the application.";
    }

    @Override
    public @NotNull final String getName() {
        return "token";
    }

    @Override
    public final IOption<String> init() {
        this.parser.accepts(getName()).withRequiredArg().ofType(String.class).defaultsTo("token_not_found").describedAs(getDescription());

        return this;
    }
}

File: src/main/kotlin/com/ryderbelserion/api/options/types/TokenOption.kt
package com.ryderbelserion.api.options.types

import com.ryderbelserion.api.options.interfaces.IOption
import joptsimple.OptionParser
import joptsimple.OptionSet
import java.util.Optional

class TokenOption(private val parser: OptionParser, override val description: String, override val name: String) : IOption<String> {

    override fun getValue(option: OptionSet): Optional<String> {
        return Optional.ofNullable(option.valueOf(this.name).toString())
    }

    override fun init(): IOption<String> {
        this.parser.accepts(this.name).withRequiredArg().ofType(String::class.java)
            .defaultsTo("token_not_found").describedAs(this.description)

        return this
    }
}